addons/base_import/models/base_import.py

```diff
@@ -626,6 +637,21 @@ def _parse_import_data(self, data, import_fields, options):
                 # We should be able to manage both case
                 index = import_fields.index(name)
                 self._parse_float_from_data(data, index, name, options)
+            elif field['type'] == 'binary' and name in import_fields:
+                index = import_fields.index(name)
+                for line in data:
+                    url_regex = re.compile(
+                        r'^ftp|http|https|smb?://'  # http:// or https:// or ftp:// or smb://
+                        r'(?:/?|[/?]\S+)$', re.IGNORECASE)
+                    if url_regex.match(line[index]):
+                        try:
+                            line[index] = self.binary2base64(line[index])
```

Comment:
Since we are potentially going to fetch multiple URLs per row, please use [a Session object](http://docs.python-requests.org/en/master/user/advanced/) so we can leverage connection pools & persistent connections.